crates/wasi-component-loader/src/http_client.rs

```diff
@@ -0,0 +1,362 @@
+use crate::{
+    names::{HTTP_CLIENT_EXECUTE_FUNCTION, HTTP_CLIENT_EXECUTE_MANY_FUNCTION, HTTP_CLIENT_RESOURCE},
+    state::WasiState,
+};
+use futures::FutureExt;
+use grafbase_telemetry::{
+    metrics::meter_from_global_provider,
+    otel::opentelemetry::{metrics::Histogram, KeyValue},
+};
+use http::{HeaderName, HeaderValue};
+use std::{
+    future::Future,
+    str::FromStr,
+    sync::LazyLock,
+    time::{Duration, Instant},
+};
+use tracing::{field::Empty, info_span, Instrument};
+use wasmtime::{
+    component::{ComponentType, Lift, LinkerInstance, Lower, ResourceType},
+    StoreContextMut,
+};
+
+static HTTP_CLIENT: LazyLock<reqwest::Client> = LazyLock::new(reqwest::Client::new);
+
+static REQUEST_METRICS: LazyLock<Histogram<u64>> = LazyLock::new(|| {
```

Comment:
Not a fan of that static implicitly relying on a global setup so deep down. All of this should be provided by the ComponentLoader with an `Arc` instead of a deep static lock hiding dependencies. Same for the http client, no need for a static lock. Just setup one in the ComponentLoader and clone it.